### PR TITLE
Create device detail page and integaret with all components

### PR DESCRIPTION
**Pull request by ajaypratap003**
### Type of change

- Enhancement / new feature

### Description
1. Create device detail page
2. Integrate Device header, DeviceInfo, and ConfigurationInfo page
3. Create routes for device detail page and children routes for DeviceInfo and ConfigurationInfo page
4. Create schema and query for device detail header

image

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ]  Resolve all warnings
- [x]  Update license
- [ ]  Write test cases for null checks in Unit Tests
- [ ] Update documentation

### CODE REVIEW

File: console/console-init/ui/src/Routes.tsx [hunk 1]
```diff
@@ -39,5 +41,10 @@ export const AppRoutes = () => (
       getComponent={getConnectionDetail}
       exact={true}
     />
+    <LazyRoute
+      path="/iot-projects/:namespace/:projectname/:deviceid/:subList"
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/Routes.tsx:45]: please change the path to `/iot-projects/:namespace/:projectname/devices/:deviceid/:subList`

File: console/console-init/ui/src/components/AdapterList/AdapterList.tsx [hunk 2]
```diff
@@ -49,7 +49,7 @@ const Adapter: React.FC<{ adapter: IAdapter }> = ({ adapter }) => {
   return (
     <>
       <Title size="xl" headingLevel="h2">
-        {type.toUpperCase() + " Adapter"}
+        {type && type.toUpperCase() + " Adapter"}
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/components/AdapterList/AdapterList.tsx:52]: we can avoid for undefined check with && and replace the logic of `type && type.toUpperCase()` with `type?.toUpperCase()`

File: console/console-init/ui/src/modules/iot-device-detail/DeviceDetailPage.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      <Breadcrumb>
+        <BreadcrumbItem>
+          <Link id="cdetail-link-home" to={"/"}>
+            Home
+          </Link>
+        </BreadcrumbItem>
+        <BreadcrumbItem>
+          <Link
+            id="device-detail-link"
+            to={`/iot-project/${namespace}/${projectname}/devices`}
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/iot-device-detail/DeviceDetailPage.tsx:49]: path will be `/iot-projects/${namespace}/${projectname}/devices`

File: console/console-init/ui/src/modules/iot-device-detail/Routes.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+const getDeviceInfoPage = () =>
+  import("modules/iot-device-detail/DeviceInfoPage");
+const getConfigurationInfoPage = () =>
+  import("modules/iot-device-detail/ConfigurationInfoPage");
+
+export const Routes = () => (
+  <SwitchWith404>
+    <Redirect path="/" to="/iot-project" exact={true} />
+    <LazyRoute
+      path="/iot-projects/:namespace/:projectname/:deviceid/deviceinfo"
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/iot-device-detail/Routes.tsx:19]: please change the path to `/iot-projects/:namespace/:projectname/devices/:deviceid/device-info`

File: console/console-init/ui/src/modules/iot-device-detail/Routes.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      getComponent={getDeviceInfoPage}
+      exact={true}
+    />
+    <LazyRoute
+      path="/iot-projects/:namespace/:projectname/:deviceid/configurationinfo"
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/iot-device-detail/Routes.tsx:24]: please change the path to `/iot-projects/:namespace/:projectname/devices/:deviceid/configuration-info`

File: console/console-init/ui/src/components/UpdatePassword/UpdatePassword.tsx [hunk 6]
```diff
@@ -16,7 +16,7 @@ import {
 import { useStoreContext, types } from "context-state-reducer";
 import { PasswordInputFieldWithToggle } from "components";
 
-export const UpdatePassword: React.FC<{}> = () => {
+export const UpdatePassword = () => {
```
Comment (rkpattnaik780) [inline console/console-init/ui/src/components/UpdatePassword/UpdatePassword.tsx:19]: Any particular reason of not having this variable as a react component?

Comment (ajaypratap003) [reply console/console-init/ui/src/components/UpdatePassword/UpdatePassword.tsx:19]: Reason is we don't have interface here, so we don't need React.FC as such. There will be no logical effect for adding or not adding.

**Comment by ajaypratap003** [comment]
> LGTM, although, I found a couple of warnings, can you please try and eliminate the ones you can?

@anukritijha , yes sure, I'll try to reduce warning as much as possible but time being we can avoid to focus on warnings and will fix all the warning after integrating mock server.

File: console/console-init/ui/mock-console-server/mock-console-server.js [hunk 7]
```diff
@@ -2128,23 +2128,48 @@ createIotDevice("iotProjectFrance", {
...
   credentials: []
 });
 
 createIotDevice("iotProjectIndia", {
   deviceId: "20",
   enabled: true,
   viaGateway: false,
-  jsonData: JSON.stringify({ ext: { ocean: "indian" } }),
+  jsonData: JSON.stringify({
+    via: ["20", "21"],
```
Comment (jbtrystram) [inline console/console-init/ui/mock-console-server/mock-console-server.js:2143]: you can't have "viaGateway" set to false and have devices in the via field at the same time

File: console/console-init/ui/mock-console-server/mock-console-server.js [hunk 8]
```diff
@@ -2128,23 +2128,48 @@ createIotDevice("iotProjectFrance", {
...
   credentials: []
 });
 
 createIotDevice("iotProjectIndia", {
   deviceId: "21",
   enabled: true,
   viaGateway: true,
-  jsonData: JSON.stringify({ via: 20, ext: { summit: "Kanchenjunga" } }),
+  jsonData: JSON.stringify({
+    via: ["device-3", "device-4"],
```
Comment (jbtrystram) [inline console/console-init/ui/mock-console-server/mock-console-server.js:2170]: same as above. Either remove the via devices or set `viaGateway` to false.

File: console/console-init/ui/mock-console-server/mock-console-server.js [hunk 9]
```diff
@@ -2177,11 +2202,52 @@ setCredentials(
...
+    {
+      type: "hashed-password",
+      "auth-id": "alternate-user-1",
+      enabled: true,
+      secrets: [
+        {
+          comment: "was just for testing"
+        }
+      ]
+    },
```
Comment (jbtrystram) [inline console/console-init/ui/mock-console-server/mock-console-server.js:2228]: you need to add a pwd-hash in the secret

File: console/console-init/ui/mock-console-server/mock-console-server.js (hunk 7 above)
Comment (ajaypratap003) [reply console/console-init/ui/mock-console-server/mock-console-server.js]: yes, I made changes you suggested.

File: console/console-init/ui/mock-console-server/mock-console-server.js (hunk 8 above)
Comment (ajaypratap003) [reply console/console-init/ui/mock-console-server/mock-console-server.js]: yes, I made changes you suggested.

File: console/console-init/ui/mock-console-server/mock-console-server.js [hunk 10]
```diff
@@ -2138,7 +2137,6 @@ createIotDevice("iotProjectFrance", {
 createIotDevice("iotProjectIndia", {
   deviceId: "20",
   enabled: true,
-  viaGateway: false,
```
Comment (jbtrystram) [inline console/console-init/ui/mock-console-server/mock-console-server.js:2138]: the viagateway field should still exist, but set to true.

File: console/console-init/ui/mock-console-server/mock-console-server.js [hunk 11]
```diff
@@ -2167,7 +2165,6 @@ createIotDevice("iotProjectIndia", {
   enabled: true,
   viaGateway: true,
   jsonData: JSON.stringify({
-    via: ["device-3", "device-4"],
```
Comment (jbtrystram) [inline console/console-init/ui/mock-console-server/mock-console-server.js]: the via field should still exist, but empty.

File: console/console-init/ui/mock-console-server/mock-console-server.js [hunk 12]
```diff
@@ -2128,23 +2127,48 @@ createIotDevice("iotProjectFrance", {
 
 createIotDevice("iotProjectFrance", {
   deviceId: "12",
```
Comment (jbtrystram) [inline console/console-init/ui/mock-console-server/mock-console-server.js:2129]: enabled and viaGateway are missing here

**Pull request merged**